### PR TITLE
Some guidance on service KPIs

### PR DESCRIPTION
**Pull request by garath**
Some guidance and notes on Service KPIs. Akin to the [Product KPI](https://github.com/dotnet/arcade/blob/master/Documentation/ProductKPIs.md) document.

### CODE REVIEW

**Comment by garath** [comment]
@ilyas1974 this is the doc I mentioned yesterday.

File: Documentation/ServiceKPIs.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  - Unknown events or triggers
+  - Malformed payloads
+  - Recoverable and unrecoverable errors
+
+Document the service's telemetry. Include for each metric:
+
+- Technical detail on how the data is collected (to help provide context)
+- _Why_ the data is collected.
+- What is _good_ and _bad_ for the data? How will a monitor determine when to act?
+- What _action_ should be taken if telemetry indicates "bad"?
```
Comment (ChadNedzlek) [inline Documentation/ServiceKPIs.md]: For this one, the metrics alerter will provide a link with every alert to a wiki that people can fill in with "what to do when you see this alert".  That way stuff is nearby.  Though that alerter is currently only in Helix, and we really need to expand it a bit.

File: Documentation/ServiceKPIs.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  - Successful and failed operations
+  - Unknown events or triggers
+  - Malformed payloads
+  - Recoverable and unrecoverable errors
+
+Document the service's telemetry. Include for each metric:
+
+- Technical detail on how the data is collected (to help provide context)
+- _Why_ the data is collected.
+- What is _good_ and _bad_ for the data? How will a monitor determine when to act?
```
Comment (ChadNedzlek) [inline Documentation/ServiceKPIs.md]: This should be more than documented in the end, there should be an alert registered for the data, but maybe we can just edit when we get around to the alerting epic?

File: Documentation/ServiceKPIs.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+- What _action_ should be taken if telemetry indicates "bad"?
+
+Work with Leadership to identify any additional telemetry for current business efforts.
+
+## First Responder
+
+As GitHub issues is the primary means of work management for DNCEng, it is expected that all First Responder efforts of note be captured as a GitHub issue and added to the "First Responder" Epic. Telemetry and workload data is collected through this channel.
```
Comment (ChadNedzlek) [inline Documentation/ServiceKPIs.md]: Some editors won't appreciate these very long lines. Could we add some line breaks?

File: Documentation/ServiceKPIs.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+Customers are may initiate communication with the First Responder team through other means, such as team alias and direct communication. Substantive communication should be moved to GitHub to allow telemetery gathering, improve workflow clarity, and increase information sharing.
```
Comment (sunandabalu) [inline Documentation/ServiceKPIs.md]: [NIT] "are may" -> "may"

Comment (sunandabalu) [inline Documentation/ServiceKPIs.md]: Teams channel is another way of communication.

File: Documentation/ServiceKPIs.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+Open a new issue for any customer support effort taking more than fifteen minutes.
+
+Ensure issues are labeled appropriately.
+
+These telemetry guidelines are meant to extend but not superseed operational guidelines set in [First Responder Responsibilities](https://github.com/dotnet/core-eng/wiki/%5Bint%5D-First-Responders).
```
Comment (sunandabalu) [inline Documentation/ServiceKPIs.md]: [NIT] "superseed" -> "supersede"

File: Documentation/ServiceKPIs.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+### Metrics
+
+Name: Count of issues opened by members outside of DNCEng
```
Comment (sunandabalu) [inline Documentation/ServiceKPIs.md]: Add extra line breaks between lines, coz right now they all appear in one line.

File: Documentation/ServiceKPIs.md (hunk 4 above)
Comment (jonfortescue) [reply Documentation/ServiceKPIs.md]: Can maybe give them [this link](https://teams.microsoft.com/l/channel/19%3aafba3d1545dd45d7b79f34c1821f6055%40thread.skype/First%2520Responders?groupId=4d73664c-9f2f-450d-82a5-c2f02756606d&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)?

Comment (jonfortescue) [reply Documentation/ServiceKPIs.md]:
```suggestion
Customers may initiate communication with the First Responder team through other means, such as team alias and direct communication. Substantive communication should be moved to GitHub to allow telemetery gathering, improve workflow clarity, and increase information sharing.
```

File: Documentation/ServiceKPIs.md (hunk 5 above)
Comment (jonfortescue) [reply Documentation/ServiceKPIs.md]:
```suggestion
These telemetry guidelines are meant to extend but not supersede operational guidelines set in [First Responder Responsibilities](https://github.com/dotnet/core-eng/wiki/%5Bint%5D-First-Responders).
```

File: Documentation/ServiceKPIs.md (hunk 3 above)
Comment (garath) [reply Documentation/ServiceKPIs.md]: I've had worse experiences when docs add line breaks - some editors are smart and helpful in flowing with available space, and if the available space is smaller than the break the lines get disjoint. I don't recall using any editors recently that don't wrap (though it's not always on by default). 

If this is a team standard and you feel strongly then I'm fine adding breaks.

File: Documentation/ServiceKPIs.md (hunk 4 above)
Comment (garath) [reply Documentation/ServiceKPIs.md]: Thanks for the link!

File: Documentation/ServiceKPIs.md (hunk 2 above)
Comment (garath) [reply Documentation/ServiceKPIs.md]: Agreed. We should look at alerting specifically, but we can certainly start musing here what the separation between "metrics" and "alerts" should be.

Comment (garath) [reply Documentation/ServiceKPIs.md]: Some thoughts. Alerts are:
 - Emergent (signaled as early as feasible, and when there is still time to respond)
 - Actionable (with a defined plan)
 - Signals of a serious threat to service stability

File: Documentation/ServiceKPIs.md (hunk 6 above)
Comment (garath) [reply Documentation/ServiceKPIs.md]: Oh weird. I missed that.

File: Documentation/ServiceKPIs.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+Telemetry on the health of a service should be considered when the service is designed.
+
+For Azure services, Application Insights is a preferred channel to report telemetry.
+
+As general guidance, consider:
+
+- Telemetry related to the _value_ the service provides to its users, example:
+  - Number of events or triggers
+  - Queries of a particular payload
```
[the remark this answers is not in the document]
Comment (garath) [reply Documentation/ServiceKPIs.md]: Fair. I've changed the wording.

File: Documentation/ServiceKPIs.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+## Mission
+
+Contribute to the successful release of .NET Core 3.0 by improving developer productivity through efficient build/test iterations
+
+## New Service Guidelines
+
+Telemetry on the health of a service should be considered when the service is designed.
+
+For Azure services, Application Insights is a preferred channel to report telemetry.
```
[the remark this answers is not in the document]
Comment (garath) [reply Documentation/ServiceKPIs.md]: I've added more to that section. My intent was to say that AI is Good and Easy, but any reliable and available solution is acceptable.

File: Documentation/ServiceKPIs.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
+# Service KPIs and Metrics
+
+## Mission
+
+Contribute to the successful release of .NET Core 3.0 by improving developer productivity through efficient build/test iterations
+
+## New Service Guidelines
+
+Telemetry on the health of a service should be considered when the service is designed.
```
[the remark this answers is not in the document]
Comment (garath) [reply Documentation/ServiceKPIs.md]: I effected change of many words.

File: Documentation/ServiceKPIs.md (hunk 1 above)
Comment (garath) [reply Documentation/ServiceKPIs.md]: Good note. 

Alerting is just out of scope for what I intended on this PR but I will start preparing another with those changes so we can start the conversation.

**Pull request merged**